src/Microsoft.ML.Transforms/StatefulFilterTransform.cs

```diff
@@ -177,14 +178,7 @@ public Cursor(StatefulFilterTransform<TSrc, TDst, TState> parent, RowCursor<TSrc
                 var appendedDataView = new DataViewConstructionUtils.SingleRowLoopDataView<TDst>(parent.Host, _parent._addedSchema);
                 appendedDataView.SetCurrentRowObject(_dst);
 
-                Func<int, bool> appendedPredicate =
-                    col =>
-                    {
-                        col = _parent._bindings.AddedColumnIndices[col];
-                        return predicate(col);
-                    };
-
-                _appendedRow = appendedDataView.GetRowCursor(appendedPredicate);
+                _appendedRow = appendedDataView.GetRowCursor(_parent._bindings.Schema);
```

Comment:
>_parent._bindings.Schema [](start = 61, length = 24)

fix to contains only columns for predicate true

Reply:
Double-check this is right. 

---
In reply to: [246108558](https://github.com/dotnet/machinelearning/pull/1975#discussion_r246108558) [](ancestors = 246108558)